app/Error/LoggerFactory.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Silex\Error;
+
+use Monolog\Handler\StreamHandler;
+use Monolog\Logger;
+use Sentry\Monolog\Handler as SentryHandler;
+use Sentry\State\HubInterface;
+use Silex\Application;
+
+final class LoggerFactory
+{
+    public static function create(
+        Application $app,
+        string $fileNameWithoutSuffix,
+        ?string $loggerName = null
```

Comment:
Can we work with just one parameter?
Currently I found on instance that has both parameters
```
$logger = LoggerFactory::create($app, 'amqp', 'amqp.event_bus_forwarder');
```
I guess this is to have one file for all AMQP messages but a dedicated channel for the `event_bus_forwarder`?

Reply:
So far this was the only one that used the extra `$loggerName` I think, but I found 15 more usages of `new Logger()` that I overlooked earlier because they weren't decorated with the `SentryPsrLoggerDecorator`. And some of those, located here, use the same StreamHandler but a different logger name. So they log to the same file but with a different logger name prefixed in the logs. For example:

```
root@udb3-web01-test:~# cat /var/www/udb-silex/log/udb2.log.1
[2021-02-09 16:34:06] udb2-events-cdbxml-enricher.DEBUG: retrieving cdbxml from http://test.open.uitdatabank.be/api/v3/event/8c76fbf0-1923-4d57-9ce5-09dcae2cdc64 [] []
[2021-02-09 16:34:15] udb2-events-cdbxml-enricher.DEBUG: sendRequest took 8978 ms. [] []
[2021-02-09 16:34:15] udb2-events-cdbxml-enricher.DEBUG: retrieved cdbxml [] []
[2021-02-09 16:34:15] udb2-events-cdbxml-enricher.DEBUG: cdbxml is valid [] []
[2021-02-09 16:34:15] udb2-events-cdbxml-enricher.DEBUG: found event in cdbxml [] []
[2021-02-09 16:34:15] udb2-events-to-udb3-event-applier.INFO: No existing offer with the same id found so it is safe to import. {"offer-id":"8c76fbf0-1923-4d57-9ce5-09dcae2cdc64"} []
[2021-02-09 16:34:16] udb2-events-to-udb3-event-applier.INFO: Offer succesfully created. {"offer-id":"8c76fbf0-1923-4d57-9ce5-09dcae2cdc64"} []
```

I'm assuming that this makes grepping etc easier, so I would keep the option to use a different logger name than the file name? But in most cases they can be the same (1 file and 1 logger) so that's why I made it optional.

Reply:
Yeah, the grouping can be nice, so indeed keep it.